src/promptflow-tools/promptflow/tools/yamls/open_source_llm.yaml

```diff
@@ -8,17 +8,27 @@ promptflow.tools.open_source_llm.OpenSourceLLM.call:
   function: call
   inputs:
     endpoint_name:
-      default: "-- please enter an endpoint name --"
       type:
       - string
+      dynamic_list:
+        func_path: promptflow.tools.open_source_llm.EndpointsContainer.list_endpoint_names
+      allow_manual_entry: false
+      is_multi_select: false
     connection:
       default: null
       type:
       - CustomConnection
     deployment_name:
-      default: null
       type:
       - string
+      dynamic_list:
+        func_path: promptflow.tools.open_source_llm.EndpointsContainer.list_deployment_names
```

Comment:
In first version of dynamic list support (which we are releasing this week), class/instance method is not supported. Please kindly change to use function. See this as an example https://github.com/microsoft/promptflow/blob/21ef07ea35fb101b11b845c22f4cbeb096fb5760/examples/tools/tool-package-quickstart/my_tool_package/tools/tool_with_dynamic_list_input.py#L5